src/main/java/com/epam/controller/TraineeController.java

```diff
@@ -0,0 +1,133 @@
+import com.epam.dto.trainer.TrainerProfile;
+import com.epam.mappers.TraineeMapper;
+import com.epam.model.Trainee;
+import com.epam.model.Trainer;
+import com.epam.model.Training;
+import com.epam.service.TraineeService;
+import com.epam.service.TrainerService;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.transaction.Transactional;
+import javax.validation.Valid;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/trainee")
+@RequiredArgsConstructor
+public class TraineeController {
+    private final TraineeMapper mapper;
+    private final TraineeService traineeService;
+    private final TrainerService trainerService;
+
+    @PostMapping
+    public ResponseEntity<TraineeRegistrationResponse> registerTrainee(@RequestBody @Valid TraineeRegistrationRequest request) {
+        Trainee trainee = mapper.traineeRegistrationRequestToTrainee(request);
+        Trainee createdTrainee = traineeService.createTrainee(trainee);
+        TraineeRegistrationResponse response = new TraineeRegistrationResponse(
+                createdTrainee.getUser().getUsername(),
+                createdTrainee.getUser().getPassword());
+        return ResponseEntity.ok(response);
+    }
+
+    @GetMapping
+    @Transactional
+    public ResponseEntity<TraineeProfileResponse> getTraineeProfile(@RequestParam String username) {
+        Optional<Trainee> traineeOptional = traineeService.getTraineeByUsername(username);
+        if (traineeOptional.isEmpty()) {
+            ResponseEntity.notFound();
+        }
+        Trainee trainee = traineeOptional.get();
+        TraineeProfileResponse profile = mapper.traineeToTraineeProfileResponse(trainee);
+
+        return ResponseEntity.ok(profile);
+    }
+
+    @PutMapping
+    @Transactional
+    public ResponseEntity<TraineeUpdateResponseDto> updateTraineeProfile(@RequestParam(name = "username") String username,
+                                                                         @Valid @RequestBody TraineeUpdateRequestDto request) {
+        Optional<Trainee> updatedTraineeOptional = traineeService.getTraineeByUsername(username);
+        if (updatedTraineeOptional.isEmpty()) {
+            return ResponseEntity.notFound().build();
+        }
+        Trainee updatedTrainee = updatedTraineeOptional.get();
+        mapper.updateEntity(updatedTrainee, request);
+
+        Trainee trainee = traineeService.updateTrainee(updatedTrainee).get();
+        TraineeUpdateResponseDto updateResponseDto = mapper.traineeToTraineeUpdateResponseDto(trainee);
+        return ResponseEntity.ok(updateResponseDto);
+    }
+
+    @DeleteMapping
+    public ResponseEntity<String> deleteTraineeProfile(@RequestParam String username) {
+        boolean deleted = traineeService.deleteTraineeByUsername(username);
```

Comment:
Here I would recommend to have a try-catch section which could catch some Exception thrown from db if something goes wrong. Based on this exception you can throw another one, like CannotDeleteTraineeException, which in turn will be called in a higher `@ExceptionHandler` and will return an appropriate text to the real user.